update-css/sw.js

```diff
@@ -0,0 +1,108 @@
+var cacheName = 'update-css-2015.1015.1202';
+
+self.addEventListener('install', function(event) {
+  self.skipWaiting();
+
+  // TODO: Comment explaining what would happen here in a real site
+
+  event.waitUntil(caches.open(cacheName).then(function(cache) {
+    return cache.addAll([
+      'style-1.css',
+      'style-2.css',
+    ]).then(getCurrentCSSFilename).then(function(filename) {
+      // Cache whichever CSS file we currently want to serve as
+      // 'style.css'
+      return cache.match(filename).then(function(response) {
+        if (response) {
+          cache.put('style.css', response);
+        }
+      });
+    });
+  }));
+});
+
+self.addEventListener('activate', function() {
+  if (self.clients && self.clients.claim) {
+    self.clients.claim();
+  }
+});
+
+self.addEventListener('fetch', function(event) {
+  // Only process fetch requests for 'style.css'. Let everything
+  // else be handled normally.
+  if (event.request.url.lastIndexOf('style.css') === -1) {
+    return;
+  }
+
+  // Respond with the cached version of 'style.css' after initiating
+  // a network fetch to update our cache if necessary.
+  event.respondWith(caches.open(cacheName).then(function(cache) {
+    // Perform a network fetch of 'style.css' to see if there's an updated
+    // version available. If there is, replace our cached version with the
+    // updated one and use `postMessage` to notify the page that there
+    // is updated CSS available.
+    simulateStyleCSSNetworkFetch().then(function(response) {
+      // In real production code, we would use smarter heuristics to
```

Comment:
When checking the ETag (or any other _smart check_ as you say) I think it is very important to do this only **without completing** the request (if you have already completed the request, simply cache the new copy). As there is no cancellable fetchs (yet), it should be convenient to make a `HEAD` fetch. Does it make sense?